src/group_theory/nilpotent.lean

```diff
@@ -460,19 +490,210 @@ begin
+  refine lower_central_series_succ_eq_bot (le_trans ((map_eq_bot_iff _).mp _) hf1),
+  apply eq_bot_iff.mpr,
+  apply (le_trans (lower_central_series.map f _)),
+  rw hn,
+  exact (le_refl _),
+end
+
+end classical
+
+/-- The range of a surejctive homomorphism from a nilpotent group is nilpotent -/
+lemma nilpotent_of_surjective
+  {G' : Type*} [group G'] {f : G →* G'} (hf : function.surjective f) [h : is_nilpotent G] :
+  is_nilpotent G' :=
+begin
+  unfreezingI { rcases h with ⟨n, hn⟩ },
+  use n,
+  apply eq_top_iff.mpr,
+  calc ⊤ = f.range : symm (monoid_hom.range_eq_top_of_surjective hf)
+    ... = subgroup.map f ⊤ : monoid_hom.range_eq_map _
+    ... = subgroup.map f (upper_central_series G n) : by rw hn
+    ... ≤ upper_central_series G' n : upper_central_series.map hf n,
+end
+
+/-- The nilpotency class of the range of a surejctive homomorphism from a
+nilpotent group is less or equal the nilpotency class of the domain -/
+lemma nilpotency_class_le_of_surjective
+  {G' : Type*} [group G'] {f : G →* G'} (hf : function.surjective f) [h : is_nilpotent G] :
+  @group.nilpotency_class G' _ (nilpotent_of_surjective hf) ≤
+    group.nilpotency_class G :=
+begin
+  apply nat.find_mono,
+  intros n hn,
+  apply eq_top_iff.mpr,
+  calc ⊤ = f.range : symm (monoid_hom.range_eq_top_of_surjective hf)
+    ... = subgroup.map f ⊤ : monoid_hom.range_eq_map _
+    ... = subgroup.map f (upper_central_series G n) : by rw hn
+    ... ≤ upper_central_series G' n : upper_central_series.map hf n,
+end
+
+/-- A quotient of a nilpotent group is nilpotent -/
+instance nilpotent_quotient_of_nilpotent (H : subgroup G) [H.normal] [h : is_nilpotent G] :
+  is_nilpotent (G ⧸ H) :=
+ nilpotent_of_surjective (show function.surjective (quotient_group.mk' H), by tidy)
+
+/-- The nilpotency class of a quotient of `G` is less or equal the nilpotency class of `G` -/
+lemma nilpotency_class_quotient_le (H : subgroup G) [H.normal] [h : is_nilpotent G] :
+  group.nilpotency_class (G ⧸ H) ≤ group.nilpotency_class G := nilpotency_class_le_of_surjective _
+
+lemma comap_comap_center {H₁ : subgroup G} [H₁.normal] {H₂ : subgroup (G ⧸ H₁)} [H₂.normal] :
+  comap (mk' H₁) (comap (mk' H₂) (center ((G ⧸ H₁) ⧸ H₂))) =
+    comap (mk' (comap (mk' H₁) H₂)) (center (G ⧸ comap (mk' H₁) H₂)) :=
+begin
+  ext,
+  simp,
+  repeat { rw mem_center_iff },
+  split; intros hx,
+  { intro y,
+    obtain ⟨y,rfl⟩ := quotient.surjective_quotient_mk' y,
+    specialize hx (quotient.mk' (quotient.mk' y)),
+    apply eq_iff_div_mem.mpr,
+    simp,
+    have hx' := eq_iff_div_mem.mp hx, clear hx,
+    apply hx',
+    apply_instance, },
+  { intro y,
+    obtain ⟨y,rfl⟩ := quotient.surjective_quotient_mk' y,
+    obtain ⟨y,rfl⟩ := quotient.surjective_quotient_mk' y,
+    specialize hx (quotient.mk' y),
+    apply eq_iff_div_mem.mpr,
+    have hx' := eq_iff_div_mem.mp hx, clear hx,
+    simp at hx',
+    apply hx',
+    apply_instance, }
+end
```

Comment:
Staring closer at it, I guess `(G ⧸ H₁) ⧸ H₂)` and `(G ⧸ comap (mk' H₁) H₂)` are isomorphic (Noethers theorem, although that’s phrased using `map`, not `comap`). We don’t have a lemma that relates the `center` of isomorphic groups yet, and even then it seems it might take a bunch of ugly calculations to relate that isomorphism to the `comap`-chains needed here.